### PR TITLE
41 drawing of simple shapes

### PR DESCRIPTION
**Pull request by JesseSwildens**

### CODE REVIEW

File: app/Core/Inc/API_simpleShapes.h
Comment (tomseli) [inline app/Core/Inc/API_simpleShapes.h:1]: API_simpleShapes does not adhere to snake_case

Comment (Mirkoet) [reply app/Core/Inc/API_simpleShapes.h:1]: agree, minor spelling mistake. REQUEST CHANGES

File: app/Core/Src/API_simpleShapes.cpp [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    if (OUTSIDE(radius, 0, MAX_RADIUS_SIZE))
+    {
+        log_message("Radius is outside of allowed range");
+        return -1;
+    }
+
+    for (int i = 0; i < MAX_CIRCLE_POINTS; i++)
+    {
+        float angle = i * 2 * (PI / 100);
+        API_setPixel(x + (cosf(angle) * radius), y + (sinf(angle) * radius), (uint8_t)color);
```
Comment (Mirkoet) [inline app/Core/Src/API_simpleShapes.cpp:153]: snake_case :)

File: app/Core/Src/API_simpleShapes.cpp [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+ * @param y y position on screen
+ * @param color color to set the pixel to
+ *
+ * @return None
+ */
+static void API_setPixel(int x, int y, uint8_t color)
+{
+    if (OUTSIDE(x, 0, VGA_DISPLAY_X) || OUTSIDE(y, 0, VGA_DISPLAY_Y))
+        return;
+    VGA_RAM1[(y * VGA_DISPLAY_X) + x] = color;
```
Comment (Mirkoet) [inline app/Core/Src/API_simpleShapes.cpp:205]: `VGA_RAM1[(y * (VGA_DISPLAY_X + 1)) + x] = color;`

Comment (Mirkoet) [reply app/Core/Src/API_simpleShapes.cpp:205]: otherwise its not a circle

File: app/Core/Src/API_simpleShapes.cpp [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+    if (OUTSIDE(radius, 0, MAX_RADIUS_SIZE))
+    {
+        log_message("Radius is outside of allowed range");
+        return -1;
+    }
+
+    for (int i = 0; i < MAX_CIRCLE_POINTS; i++)
+    {
+        float angle = i * 2 * (PI / 100);
```
Comment (Mirkoet) [inline app/Core/Src/API_simpleShapes.cpp:152]: enable the FPU, maybe easier if it is done in this commit
`SCB->CPACR |= ((3UL << 10 * 2) | (3UL << 11 * 2));`

File: app/Core/Src/API_simpleShapes.cpp [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+ * @brief Drawing circle
+ *
+ * @param x_y position on screen
+ * @param radius Radius of the circle
+ * @param color Color of the circle
+ * @param reserved Currently unused. Reserved for future use.
+ *
+ * @return -1 if error occured, otherwise 0
+ */
+int API_draw_circle(int x, int y, int radius, int color, int reserved)
```
Comment (Mirkoet) [inline app/Core/Src/API_simpleShapes.cpp:128]: This function is limited to a radius of 25 (which is really rather small). Bresenhams algorithm can be used to fix this issue

**Review by Mirkoet — CHANGES_REQUESTED** [review]
Really minor spelling things, i cant really find anything else

**Review by tomseli — APPROVED** [review]
lgtm 👍

**Review by Mirkoet — APPROVED** [review]
The drawing functions work. for mvp this is enough.

**Pull request merged**